test/e2e/ha/command_test.go

```diff
@@ -0,0 +1,369 @@
+package ha
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	core "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	contrail "github.com/Juniper/contrail-operator/pkg/apis/contrail/v1alpha1"
+	"github.com/Juniper/contrail-operator/pkg/client/keystone"
+	"github.com/Juniper/contrail-operator/pkg/client/kubeproxy"
+	"github.com/Juniper/contrail-operator/pkg/controller/utils"
+	"github.com/Juniper/contrail-operator/test/logger"
+	"github.com/Juniper/contrail-operator/test/wait"
+)
+
+func TestHACommand(t *testing.T) {
```

Comment:
Why is this test a part of HA test suite when it only test single replica and upgrade?

Reply:
I think that upgrade test of single instance should be covered by existing TestCommandServices. We already create all required resources there. Now we are doing the same stuff twice and it consume a lot of time.

I strongly suggest add upgrade step to existing test instead of creating a new one.